src/lib/Libnet/hnls.c

```diff
@@ -568,3 +570,140 @@ free_if_info(struct log_net_info *ni)
 		free(temp);
 	}
 }
+
+/**
+* @brief
+	Get a list of all IPs
+*
+* @return 
+*		Comma separated list of ips in string format
+* 
+* @par Side Effects:
+*	None
+*
+* @par MT-safe: Yes
+*
+*/
+char *
+get_all_ips(char *msg)
+{
+	char * nodenames = NULL;
+#if defined(linux)
+	char *tmp;
+	int len, hlen, ret;
+	struct ifaddrs *ifp, *listp;
+	char buf[80] = {'\0'};
+	char *p;
+
+	msg[0] = '\0';
+
+	ret = getifaddrs(&ifp);
+
+	if ((ret != 0) || (ifp == NULL)) {
+		strncpy(msg, "Failed to obtain interface names", 2048);
+		return NULL;
+	}
+
+	len = 0;
+	for (listp = ifp; listp; listp = listp->ifa_next) {
+		if ((listp->ifa_addr == NULL) || (listp->ifa_addr->sa_family != AF_INET)) continue;
+		sprintf(buf, "%s", netaddr((struct sockaddr_in *)listp->ifa_addr));
+		if (!strcmp(buf,"unknown")) continue;
+		if ((p=strchr(buf, ':')))
+			*p = '\0';
+
+		hlen = strlen (buf);
+		tmp = realloc(nodenames, len + hlen + 2); /* 2 for comma and null char */
+		if (!tmp) {
+			strncpy(msg, "Out of memory", LOG_BUF_SIZE - 1);
+			free(nodenames);
+			nodenames = NULL;
+			break;
+		}
+		nodenames = tmp;
+
+		if (len == 0)
+			strcpy(nodenames, buf);
+		else {
+			strcat(nodenames, ",");
+			strcat(nodenames, buf);
+		}
+		len += hlen + 2;
+	}
+
+	freeifaddrs(ifp);
+
+#elif defined(WIN32)
+
+	int i, len=0, hlen, ret;
+	char *tmp;
+	char buf[80] = {'\0'};
+	
+	/* Variables used by GetIpAddrTable */
+	PMIB_IPADDRTABLE pIPAddrTable;
+	DWORD dwSize = 0;
+	DWORD dwRetVal = 0;
+	IN_ADDR IPAddr;
+
+	/* Variables used to return error message */
+	LPVOID lpMsgBuf;
+
+	// Before calling AddIPAddress we use GetIpAddrTable to get
+	// an adapter to which we can add the IP.
+	pIPAddrTable = (MIB_IPADDRTABLE *) malloc(sizeof (MIB_IPADDRTABLE));
+
+	if (pIPAddrTable) {
+		// Make an initial call to GetIpAddrTable to get the
+		// necessary size into the dwSize variable
+		if (GetIpAddrTable(pIPAddrTable, &dwSize, 0) ==
+			ERROR_INSUFFICIENT_BUFFER) {
+			free(pIPAddrTable);
+			pIPAddrTable = (MIB_IPADDRTABLE *) malloc(dwSize);
+
+		}
+		if (pIPAddrTable == NULL) {
+			strncpy(msg, "Memory allocation failed for GetIpAddrTable", LOG_BUF_SIZE - 1);
+			exit(1);
```

Comment:
Is it better to return an error code instead of exiting? Also the error messages must get logged in the log file somehow.

Reply:
Yes, you are right, It's better to return instead of exit().
I have updated the code to return in case of error and the error message would get logged in via caller of this function. 
Please have a look and share your review comments, Thanks